test/box-luatest/gh_7901_iproto_request_handlers_overriding_test.lua

```diff
@@ -0,0 +1,307 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new{
+        alias   = 'dflt',
+    }
+    cg.server:start()
+
+    cg.server:exec(function(net_box_uri)
+        local uri = require('uri')
+        local socket = require('socket')
+        local t = require('luatest')
+
+        -- Connect to the server.
+        local u = uri.parse(net_box_uri)
+        local s = socket.tcp_connect(u.host, u.service)
+        t.assert_is_not(s, nil)
+        -- Skip the greeting.
+        t.assert_equals(#s:read(128), 128)
+        rawset(_G, 's', s)
+    end, {cg.server.net_box_uri})
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Checks that `box.iproto.override` works correctly.
+g.test_box_iproto_override = function(cg)
```

Comment:
Nit: The test is huge. Can you split it in smaller test cases?

Reply:
Fixed: refactored the test.